patches/api/0347-Schoolable-Fish-API.patch

```diff
@@ -0,0 +1,92 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Tue, 7 Dec 2021 19:50:01 -0500
+Subject: [PATCH] Schoolable Fish API
+
+
+diff --git a/src/main/java/io/papermc/paper/entity/SchoolableFish.java b/src/main/java/io/papermc/paper/entity/SchoolableFish.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..c75b9a8b5af6bf007ed94bb322f097d8bc0a0258
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/entity/SchoolableFish.java
+@@ -0,0 +1,47 @@
++package io.papermc.paper.entity;
++
++import org.bukkit.entity.Fish;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Represents a fish that can school with other fish.
++ */
++public interface SchoolableFish extends Fish {
++
++    /**
++     * Forces this fish to follow the given fish.
++     *
++     * @param leader fish to follow
++     */
++    void startFollowing(@NotNull SchoolableFish leader);
++
++    /**
++     * Causes the fish to stop following their current
++     * leader.
++     */
++    void stopFollowing();
++
++    /**
++     * Returns if this fish under normal circumstances can be followed.
++     *
++     * @return can be followed or not
++     */
++    boolean canBeFollowed();
```

Comment:
seems like the implementation for this will return false when a fish has no followers? also might want to expose the max school size

Reply:
Correct, it's like this because it determines if other fish are able to follow that fish's school. 
![image](https://user-images.githubusercontent.com/23108066/147779255-99b83e0d-c770-4cdf-86b2-d09d5235e92f.png)
It's used in this task, where if the fish doesn't have a leader or can be followed it will try to follow it's leader.

 Also exposed max school size!